### DISCUSSION TITLE
How do I download and use the CLI?

### DISCUSSION THREAD

**Discussion by stevebaxter** [winder/Universal-G-Code-Sender#2596] [Help] [answered]
This may be a stupid question, but how do I download and use the CLI? We want to use it to send gcode to a CNC controller as part of some factory automation. This will initially be on Windows for testing, then moving to a Raspberry Pi for production.

I can't find `ugs-cli-2.0-SNAPSHOT.jar` anywhere in the downloads (as documented here https://github.com/winder/Universal-G-Code-Sender/tree/master/ugs-cli).

**Comment by breiler** [comment, marked as answer]
Hi Steve, no it is not a stupid question. There is not much documentation in it.

To use the CLI version you need to download the UGS classic edition. And you will need to install Java 17 on your system. There is a start script `ugs-cli.sh` for Linux but not Windows, you will need to create that yourself.

**Reply by stevebaxter** [reply]
Thanks, that's very helpful!
